actions: Add new issues to project

jodh-intel opened the pull request:
Add configuration to trigger a new GitHub action for the repository that
will add all newly created issues to the organisation-level issue
backlog project.

Fixes: #157.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

jodh-intel commented:
First of a set of PRs (one per Kata repo) that will add a github action to auto add newly created issues to the new issue backlog project.

I'll use this PR to test before raising the other PRs.

/cc @ariel-adam, @amshinde.

jodh-intel commented:
Oh - and in case you didn't guess from the flurry of issues I raised, the issues *and* this PR were all raised automatically using the awesome hub command ;)

jodh-intel commented:
This PR is using the following GitHub action:

- https://github.com/marketplace/actions/automate-projects

The source for this action is APACHE-2.0-licensed and can be found here:

- https://github.com/takanabe/github-actions-automate-projects

The code is written in golang. Although I've reviewed the code (`Dockerfile.build` and `*.go` files), please can the reviewers also check it and comment here. As the stats below show, it's a very small amount of code:

```
--------------------------------------------------------------------------------
 Language             Files        Lines        Blank      Comment         Code
--------------------------------------------------------------------------------
 JSON                     2          619            1            0          618
 Go                       4          426           69           39          318
 Markdown                 1          111           28            0           83
 Makefile                 1           29            3           12           14
 YAML                     1            9            0            0            9
 Docker                   1           12            3            0            9
--------------------------------------------------------------------------------
 Total                   10         1206          104           51         1051
--------------------------------------------------------------------------------
```

amshinde commented:
@jodh-intel I took a look at the code for the `automate-actions` code, and it looks sane to me.
As long as we use a particular tag/version for the github actions and review before we update tags/versions, we should be good.

amshinde commented on .github/workflows/add-issues-to-project.yaml:
```diff
@@ -0,0 +1,16 @@
+      uses: docker://takanabe/github-actions-automate-projects:v0.0.1
+      if: github.event_name == 'issues' && github.event.action == 'opened'
+      env:
+        GITHUB_TOKEN: ${{ secrets.KATA_GITHUB_ACTIONS_TOKEN }}
```
I suppose you have created a token here with restricted access rather than using the default GITHUB_TOKEN?

jodh-intel replied:
That's right - `KATA_GITHUB_ACTIONS_TOKEN` is a new token I've created specifically for (this) action. It's technically a personal access token my user has created on the org and then each repo has a "secret" with the value of this token. This is documented here:

- https://github.com/marketplace/actions/automate-projects#organization-based-project

jodh-intel commented:
When https://github.com/kata-containers/.github/pull/10 lands, I can rework this to make a call to the `kata-hub` script, similar to what https://github.com/kata-containers/kata-containers/pull/529 is doing.

Commit:
actions: Add new issues to project

Add configuration to trigger a new GitHub action for the repository that
will add all newly created issues to the organisation-level issue
backlog project.

Fixes: #157.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

fidencio commented:
Let me close this one as it's a do-not-merge which won't make to the 1.13.0 release.
